Web Certificates: Open edX Base & Distinguished Styling

talbs opened the pull request:
This work provides initial static rendering/styling for Open edX web-based certificate views. 

### Currently Included

* a manually copied version of UX Pattern Library assets (see ux.edx.org) at a global level
* Specific vendor code/utilities for cert rendering - Susy for grids, Domine + Font Awesome typefaces
* Responsive layout and visual styling for a base + distinguished view
* assets (fonts, sample images, same copy) for same views

- - -

### UI TODOs still include:

* i18n
* ~~RTL full support + configuration~~
* ~~Browser rendering review~~

- - -

### Screenshots (as of 5/5/2015)

**Base Certificate**
![cert-full-base](https://cloud.githubusercontent.com/assets/163763/7507518/c15d8e20-f43e-11e4-982f-d395450068e6.png)

**Distinguished Certificate**
![cert-full-distinguished](https://cloud.githubusercontent.com/assets/163763/7507520/c4e5e556-f43e-11e4-9890-a69387d70a52.png)

**Distinguished Certificate + Recipient is Logged in User**
![cert-full-distinguished user-prompt](https://cloud.githubusercontent.com/assets/163763/7507523/c7bfa1e0-f43e-11e4-8fd4-f4a8e6877017.png)

talbs commented:
FYI, @mattdrayer, here is the initial static rendering/styling work for the Open edX web certs. I've included screengrabs here and will dupe them on all of the relevant JIRA tickets. I'm going to place some "NOTES" inline for your reference and for our walkthrough tomorrow.

@marcotuts, since you know the context best, mind reviewing this both from a visual rendering and a FED perspective? I'm happy to walk through anything I'm doing and also give more context.

Shout if you folks need any help or if this PR should be against a different branch.

talbs commented:
@frrrances, FYI, in case you're interested or think this needs a second FED review.

talbs commented on lms/static/certificates/css/main-ltr.css:
```diff
@@ -0,0 +1,6 @@
+﻿@font-face{font-family:"Open Sans";font-style:normal;font-weight:300;src:url("../fonts/OpenSans/OpenSans-Light-webfont.eot?#iefix") format("embedded-opentype"),url("../fonts/OpenSans/OpenSans-Light-webfont.woff") format("woff"),url("../fonts/OpenSans/OpenSans-Light-webfont.ttf") format("truetype"),url("../fonts/OpenSans/OpenSans-Light-webfont.svg#Open Sans") format("svg")}@font-face{font-family: ...
```
NOTE: compiled asset from corresponding ``*.scss`` file.

talbs commented on lms/static/certificates/css/main-rtl.css:
```diff
@@ -0,0 +1,6 @@
+﻿@font-face{font-family:"Open Sans";font-style:normal;font-weight:300;src:url("../fonts/OpenSans/OpenSans-Light-webfont.eot?#iefix") format("embedded-opentype"),url("../fonts/OpenSans/OpenSans-Light-webfont.woff") format("woff"),url("../fonts/OpenSans/OpenSans-Light-webfont.ttf") format("truetype"),url("../fonts/OpenSans/OpenSans-Light-webfont.svg#Open Sans") format("svg")}@font-face{font-family: ...
```
NOTE: compiled asset from corresponding ``*.scss`` file.

talbs commented on lms/static/certificates/js/vendor/jquery-1.10.2.min.js:
```diff
@@ -1,6 +0,0 @@
-/*! jQuery v1.10.2 | (c) 2005, 2013 jQuery Foundation, Inc. | jquery.org/license
```
NOTE: removing unnecessary and redundant vendor files.

talbs commented on lms/static/certificates/js/vendor/modernizr-2.6.2.min.js:
```diff
@@ -1,4 +0,0 @@
-/* Modernizr 2.6.2 (Custom Build) | MIT & BSD
```
NOTE: removing unnecessary and redundant vendor files.

talbs commented on lms/static/certificates/sass/_build.scss:
```diff
@@ -1,45 +1,53 @@
 // ------------------------------
```
NOTE: revamping this shared build compile list to use static version of PL assets.

talbs replied:
Also, it sucks that Sass interpolation can't handle passing a pathing variable (like, ``$edx-pl-path``) into all of these ``@import`` statements.

talbs commented on lms/static/certificates/sass/_build.scss:
```diff
@@ -1,45 +1,53 @@
 // #SHAME
 // ------------------------------
-@import 'utilities/shame';
+@import 'overrides';
```
NOTE: overrides contains shame, developer, and other last of the compile extensions.

talbs commented on lms/static/certificates/sass/_components.scss:
```diff
@@ -0,0 +1,641 @@
+// ------------------------------
+// #OVERRIDE - COPY
+// ------------------------------
+// for some reason in Sass compilation, function calls as variables aren't playing well :/
```
NOTE: @clrux  and I need to investigate why these Sass function calls can't be passed in as variables at this level. Light redundancy is needed for now with that in mind.

talbs commented on lms/static/certificates/sass/_components.scss:
```diff
@@ -0,0 +1,641 @@
+// ------------------------------
+// #OVERRIDE - HEADINGS
+// ------------------------------
+// for some reason in Sass compilation, function calls as variables aren't playing well :/
```
NOTE: @clrux  and I need to investigate why these Sass function calls can't be passed in as variables at this level. Light redundancy is needed for now with that in mind.

talbs commented on lms/static/certificates/sass/_components.scss:
```diff
@@ -0,0 +1,641 @@
+}
+
+// ------------------------------
+// #DEPTH
```
NOTE: cheating here and doing some PL work ahead of its time with UI depth styling. :P

talbs commented on lms/static/certificates/sass/_components.scss:
```diff
@@ -0,0 +1,641 @@
+// ------------------------------
+// #IMAGES
+// ------------------------------
+// circular cropped
```
NOTE: utility for doing what them there kids are doing these days with rounded images.

clrux replied:
Them there's some nice rounded images.

talbs commented on lms/static/certificates/sass/_components.scss:
```diff
@@ -0,0 +1,641 @@
+    }
+}
+
+// main accomplishment
```
NOTE: considering a few layers of basic accomplishments with this set up - a main accomplishment (the focus of a view) vs. brief accomplishments (supplementally listed or listed in contexts of other application views - e.g. profile, progress, dashboard).

talbs commented on lms/static/certificates/sass/_components.scss:
```diff
@@ -0,0 +1,641 @@
+}
+
+// ------------------------------
+// #RENDERING
```
NOTE: breaking up the actual rendering of an accomplishment from the listing styles above. This is pretty specific to the content we're expecting with certs for now, but classes have been abstracted out.

marcotuts replied:
NOTE: I think the plan is to not have any accomplishments listed on these pages in the short term. It isn't currently a Saudi commitment and it isn't clear when we would revisit supplemental accomplishments on these views. I suppose upon reflection it doesn't hurt us to have it in the FED / styling code early. 

Q: One note about this abstraction is also that this would eventually (presumably?) move out of the certificates application context if we expect multiple parts of the platform to use it. I would expect this to move to the /openedx/ folder (our new version of common), or whatever the new PL work suggests from a file organization standpoint. 

(No changes necessary here, just commenting on the new and forward thinking work her! :100: )

talbs commented on lms/static/certificates/sass/_components.scss:
```diff
@@ -0,0 +1,641 @@
+    }
+
+    .accomplishment-type-label {
+        width: span(10);
```
NOTE: new susy grid-management syntax. ``span(*)`` is a susy function that will do grid math and give us the width of columns separated from the actual laying out of the columns. Handy for when you want control of the layout, but want the same calculations to base that off of.

See http://www.sitepoint.com/creating-responsive-grid-system-susy-breakpoint/ and http://susydocs.oddbird.net/en/latest/.

talbs commented on lms/static/certificates/sass/_components.scss:
```diff
@@ -0,0 +1,641 @@
+}
+
+// ------------------------------
+// #CERTIFICATE
```
NOTE: styling specific to the certificate accomplishment views.

talbs commented on lms/static/certificates/sass/_components.scss:
```diff
@@ -0,0 +1,641 @@
+        .accomplishment-recipient,
+        .accomplishment-course,
+         {
+            font-family: $font-family-serif;
```
NOTE: Our first serif - "that's one small type choice for this project, one giant leap for Open edX typography."

marcotuts replied:
I believe that this repository should commit itself to acsending to the goal, before this the terminal of this quarter, of landing this typographical lead-ership, to stem the tide of quality descending legacy stylings and return us serifly to the  land of clean styles we live in. 

That's about as many typographical puns I can ligature together with the spine of a moon quote to work from.

talbs commented on lms/static/certificates/sass/_components.scss:
```diff
@@ -0,0 +1,641 @@
+
+    .accomplishment-rendering {
+
+        // hide the fancy
```
NOTE: Poor man's way of visually differentiating base vs. distinguished. The DOM items noted here should be removed from the template rendering as well.

talbs commented on lms/static/certificates/sass/_layouts.scss:
```diff
@@ -0,0 +1,324 @@
+// ------------------------------
```
NOTE: this partial controls almost all of the layout controls, including responsive details.

talbs commented on lms/static/certificates/sass/_layouts.scss:
```diff
@@ -0,0 +1,324 @@
+    margin-bottom: spacing-vertical(base);
+    padding: 0 (gutter()*2);
+
+    @include susy-breakpoint($bp-medium, $susy) {
```
NOTE: new susy grid-management syntax. ``susy-breakpoint`` is a mixin that combines susy with breakpoint (http://breakpoint-sass.com/) for easy mgmt of breakpoints. ``gutter()`` is another handy susy function for getting the width of your grid's gutters out of a grid layout context.

See http://www.sitepoint.com/creating-responsive-grid-system-susy-breakpoint/ and http://susydocs.oddbird.net/en/latest/.

talbs commented on lms/static/certificates/sass/_layouts.scss:
```diff
@@ -0,0 +1,324 @@
+}
+
+%layout {
+    @include container();
```
NOTE: new susy grid-management syntax. ``container()`` sets up the parent of columns and its properties (max-width, etc.)

marcotuts replied:
This is neat! Well it isn't Neat, it's Suzy, but it's awesome.

talbs commented on lms/static/certificates/sass/_layouts.scss:
```diff
@@ -0,0 +1,324 @@
+            margin-bottom: spacing-vertical(small);
+
+            @include susy-breakpoint($bp-large, $susy) {
+                @include span(8 first);
```
NOTE: new susy grid-management syntax - ``span(*)`` is a method for laying out columns. See http://susydocs.oddbird.net/en/latest/shorthand/#span.

talbs commented on lms/static/certificates/sass/_layouts.scss:
```diff
@@ -0,0 +1,324 @@
+            }
+
+            @include susy-breakpoint($bp-medium, $susy) {
+                @include gallery(4);
```
NOTE: new susy grid-management syntax - this creates a column layout with row breaks set for nth elements based on what's passed in (in this case, every 4 elements). See - http://susydocs.oddbird.net/en/latest/toolkit/#gallery

marcotuts replied:
sweet! :100: says the peanut gallery.

talbs commented on lms/static/certificates/sass/_utilities.scss:
```diff
@@ -0,0 +1,61 @@
+// ------------------------------
+$em-base:       16; // deliberately sets bourbon-based em-base (http://bourbon.io/docs/#em-base)
+
+// certificate characteristics
```
NOTE: setting some easy styling customization points for certs.

talbs commented on lms/static/certificates/sass/_utilities.scss:
```diff
@@ -0,0 +1,61 @@
+$cert-decorative-color-emphasized: palette(grayscale-cool, mid-light);
+
+// typography: config
+@include font-face('Domine', '../fonts/Domine/Domine-Regular-webfont', 400, $file-formats: eot woff ttf svg);
```
NOTE: adding in serif typeface assets.

talbs commented on lms/static/certificates/sass/base/_anims.scss:
```diff
@@ -1,258 +0,0 @@
-// ------------------------------
```
NOTE: removing older/irrelevant Sass partial.

talbs commented on lms/static/certificates/sass/base/_fonts.scss:
```diff
@@ -1,12 +0,0 @@
-// ------------------------------
```
NOTE: removing older/irrelevant Sass partial.

talbs commented on lms/static/certificates/sass/base/_graphics.scss:
```diff
@@ -1,6 +0,0 @@
-// ------------------------------
```
NOTE: removing older/irrelevant Sass partial.

talbs commented on lms/static/certificates/sass/base/_reset.scss:
```diff
@@ -1,79 +0,0 @@
-// ------------------------------
```
NOTE: removing older/irrelevant Sass partial.

talbs commented on lms/static/certificates/sass/base/_typography.scss:
```diff
@@ -1,186 +0,0 @@
-// ------------------------------
```
NOTE: removing older/irrelevant Sass partial.

talbs commented on lms/static/certificates/sass/components/_controls.scss:
```diff
@@ -1,115 +0,0 @@
-// ------------------------------
```
NOTE: removing older/irrelevant Sass partial.

talbs commented on lms/static/certificates/sass/components/_footer.scss:
```diff
@@ -1,120 +0,0 @@
-// ------------------------------
```
NOTE: removing older/irrelevant Sass partial.

talbs commented on lms/static/certificates/sass/components/_header.scss:
```diff
@@ -1,43 +0,0 @@
-// ------------------------------
```
NOTE: removing older/irrelevant Sass partial.

talbs commented on lms/static/certificates/sass/components/_navigation.scss:
```diff
@@ -1,101 +0,0 @@
-// ------------------------------
```
NOTE: removing older/irrelevant Sass partial.

talbs commented on lms/static/certificates/sass/contexts/_ie.scss:
```diff
@@ -1,22 +0,0 @@
-// ------------------------------
```
NOTE: removing older/irrelevant Sass partial.

talbs commented on lms/static/certificates/sass/contexts/_print.scss:
```diff
@@ -1,27 +0,0 @@
-// ------------------------------
```
NOTE: removing older/irrelevant Sass partial.

talbs commented on lms/static/certificates/sass/main-ltr.css:
```diff
@@ -1,1676 +0,0 @@
-html {
```
NOTE: removing older/irrelevant compiled CSS file.

talbs commented on lms/static/certificates/sass/main-ltr.scss:
```diff
@@ -1,19 +1,24 @@
 // ------------------------------
```
NOTE: removing older/irrelevant Sass file.

talbs commented on lms/static/certificates/sass/main-rtl.css:
```diff
@@ -1,1676 +0,0 @@
-html {
```
NOTE: removing older/irrelevant compiled CSS file.

talbs commented on lms/static/certificates/sass/main-rtl.scss:
```diff
@@ -1,19 +1,24 @@
 // ------------------------------
```
NOTE: removing older/irrelevant Sass file.

talbs commented on lms/static/certificates/sass/utilities/_mixins.scss:
```diff
@@ -1,130 +0,0 @@
-// ------------------------------
```
NOTE: removing older/irrelevant Sass partial.

talbs commented on lms/static/certificates/sass/utilities/_shame.scss:
```diff
@@ -1,4 +0,0 @@
-// ------------------------------
```
NOTE: removing older/irrelevant Sass partial.

talbs commented on lms/static/certificates/sass/utilities/_variables-rtl.scss:
```diff
@@ -1,5 +0,0 @@
-// ------------------------------
```
NOTE: removing older/irrelevant Sass partial.

talbs commented on lms/static/certificates/sass/utilities/_variables.scss:
```diff
@@ -1,181 +0,0 @@
-// ------------------------------
```
NOTE: removing older/irrelevant Sass partial.

talbs commented on lms/static/certificates/sass/views/_certificate.scss:
```diff
@@ -1,150 +0,0 @@
-// ------------------------------
```
NOTE: removing older/irrelevant Sass partial.

talbs commented on lms/static/certificates/sass/views/_invalid.scss:
```diff
@@ -1,89 +0,0 @@
-// ------------------------------
```
NOTE: removing older/irrelevant Sass partial.

talbs commented on lms/static/certificates/sass/views/_valid.scss:
```diff
@@ -1,207 +0,0 @@
-// ------------------------------
```
NOTE: removing older/irrelevant Sass partial.

talbs commented on lms/static/certificates/sass/views/_validate.scss:
```diff
@@ -1,289 +0,0 @@
-// ------------------------------
```
NOTE: removing older/irrelevant Sass partial.

talbs commented on lms/templates/certificates/_accomplishment-banner.html:
```diff
@@ -0,0 +1,35 @@
+<div class="wrapper-banner wrapper-banner-user">
```
NOTE: banner displayed to a logged in user who's viewing their own accomplishment.

talbs commented on lms/templates/certificates/_accomplishment-related.html:
```diff
@@ -0,0 +1,49 @@
+<div class="wrapper-accomplishment-related">
```
NOTE: future-thinking UX here of associating other course-based accomplishments with the main one featured (certificates in this case). We can remove if not needed now.

marcotuts replied:
If we'd like to leave the partial but just not reference it in the main view that might be ok. (my 0.2 cents)

talbs commented on lms/templates/certificates/_accomplishment-rendering.html:
```diff
@@ -0,0 +1,142 @@
+<article class="accomplishment accomplishment-main">
```
NOTE: rendering of accomplishment (aka the details).

talbs commented on lms/templates/certificates/_accomplishment-rendering.html:
```diff
@@ -0,0 +1,142 @@
+                    <img class="src" src="/static/certificates/images/demo-ico-distinguished.svg" alt="">
+                </span>
+
+                <span class="accomplishment-type-label hd-4">Distinguished Certificate of Achievement</span>
```
NOTE: @mattdrayer, we'll need to abstract this value.

talbs commented on lms/templates/certificates/_accomplishment-rendering.html:
```diff
@@ -0,0 +1,142 @@
+
+            <div class="accomplishment-type">
+                <span class="accomplishment-type-symbol">
+                    <img class="src" src="/static/certificates/images/demo-ico-distinguished.svg" alt="">
```
NOTE: @mattdrayer, we'll need to abstract this value.

talbs commented on lms/templates/certificates/_accomplishment-rendering.html:
```diff
@@ -0,0 +1,142 @@
+            </div>
+
+            <div class="accomplishment-signatories">
+                <h3 class="accomplishment-signatories-title hd-6 sr-only">Noted by</h3>
```
?: Is this the right wording?

talbs commented on lms/templates/certificates/_accomplishment-rendering.html:
```diff
@@ -0,0 +1,142 @@
+                <div class="wrapper-signatories">
+                    <ul class="list-signatories">
+                        <li class="signatory">
+                            <img class="signatory-signature" src="/static/certificates/images/demo-sig1.png" alt="">
```
NOTE: @mattdrayer, we'll need to abstract these signatures.

talbs commented on lms/templates/certificates/_accomplishment-rendering.html:
```diff
@@ -0,0 +1,142 @@
+            <div class="accomplishment-orgs">
+                <h3 class="accomplishment-orgs-title hd-6 sr-only">Supported by the following organizations</h3>
+
+                <ul class="wrapper-orgs list-orgs list-orgs-4+">
```
NOTE: @mattdrayer, we'll need to abstract these organizations.

talbs commented on lms/templates/certificates/_accomplishment-rendering.html:
```diff
@@ -0,0 +1,142 @@
+            </div>
+
+            <div class="accomplishment-orgs">
+                <h3 class="accomplishment-orgs-title hd-6 sr-only">Supported by the following organizations</h3>
```
?: Is this the correct wording for this title?

talbs commented on lms/templates/certificates/_accomplishment-rendering.html:
```diff
@@ -0,0 +1,142 @@
+                    <dt class="label sr-only">Awarded to:</dt>
+                    <dd class="value copy copy-meta">
+                        <span class="recipient-img">
+                            <img class="src" src="/static/certificates/images/demo-user-profile2.png" alt="">
```
NOTE: @mattdrayer, we'll need to sync up a user's profile photo here.

talbs commented on lms/templates/certificates/_accomplishment-rendering.html:
```diff
@@ -0,0 +1,142 @@
+                        </span>
+                        <div class="recipient-details">
+                            <h3 class="recipient-name">${accomplishment_copy_name}</h3>
+                            <p class="recipient-username">usernameatedx</p>
```
NOTE: @mattdrayer, we'll need to plumb through a user's username here.

talbs commented on lms/templates/certificates/_assets-primary.html:
```diff
@@ -1 +1 @@
-<link rel="stylesheet" href="${document_stylesheet_url_application}" />
+<link rel="stylesheet" href="${document_stylesheet_url_application}">
```
HTML5 FTW!

talbs commented on lms/templates/certificates/_certificate-footer.html:
```diff
@@ -1,17 +0,0 @@
-<div class="wrapper-footer">
```
NOTE: removing unnecessary partial.

talbs commented on lms/templates/certificates/_certificate-header.html:
```diff
@@ -1,10 +0,0 @@
-<div class="wrapper-header">
```
NOTE: removing unnecessary partial.

talbs commented on lms/templates/certificates/accomplishment-base.html:
```diff
@@ -0,0 +1,40 @@
+<%! from django.utils.translation import ugettext as _ %>
```
NOTE: new base template certificates inherit from.

talbs commented on lms/templates/certificates/accomplishment-base.html:
```diff
@@ -0,0 +1,40 @@
+    <%include file="_assets-primary.html" />
+</head>
+
+<body class="layout-accomplishment view-valid-accomplishment ${dir_rtl} ${document_body_class} certificate certificate-distinguished" data-view="valid-accomplishment">
```
NOTE: @mattdrayer, we'll need to sync up the following classes here:

* ``certificate`` - added to all accomplishments that are certificates
* ``certificate`` + ``certificate-base`` - base certificate
* ``certificate`` + ``certificate-distinguished`` - distinguished certificate
* ``view-valid-accomplishment`` - if accomplishment is valid
* ``view-invalid-accomplishment`` - if accomplishment is invalid (we'll be showing the invalid.html file with specific UI here too)

talbs commented on lms/templates/certificates/certificate-base.html:
```diff
@@ -1,31 +0,0 @@
-<%! from django.utils.translation import ugettext as _ %>
```
NOTE: removing older template.

talbs commented on lms/templates/certificates/invalid.html:
```diff
@@ -6,41 +6,44 @@
 
```
NOTE: still need to confirm that this view is needed and if so, styling/rendering is still incomplete.

talbs commented on static/edx-pl/sass/_build.scss:
```diff
@@ -0,0 +1,41 @@
+// ------------------------------
```
NOTE: edX Pattern Library compile (not used directly in certs, since certs has its own compile that references shared assets).

Kelketek commented:
@talbs I'm getting `ValueError: The file 'images/bg-distinguished.png' could not be found with <pipeline.storage.PipelineCachedStorage object at 0x5686690>.` when I try to collectstatic after these changes. Am I missing something?

Kelketek commented:
@talbs I looked into this further. These two files are in the repo:

    lms/static/certificates/css/main-ltr.css
    lms/static/certificates/css/main-rtl.css

They appear to be generated files (no newlines/indentation), so I'm guessing they're not supposed to be there. They cause the `paver update_assets` command to fail.

Kelketek commented:
![cert](https://cloud.githubusercontent.com/assets/1099483/7541479/5f6c6f90-f57b-11e4-800a-051ceb0a2615.png)

After having fixed that problem, the certificate still does not render quite correctly. Attached is a screenshot of the result. I see no Javascript files being loaded. When you told me that the social bar is static, should I take it to mean that no code has been added to activate the buttons when they display correctly? I'm trying to figure out where to inject what I want to add. If no current code exists for social buttons, this would mean mine would be the first, and the first JS to be loaded into the page. But it would also mean I'll be waiting for the rest of it to be done before my code will be even worth considering.

marcotuts commented on lms/static/certificates/sass/_components.scss:
```diff
@@ -0,0 +1,641 @@
+// ------------------------------
+// #OVERRIDE - BUTTONS
+// ------------------------------
+a.btn {
```
Q / NIT -  Didn't know if the "a" was deliberate here or not, but it was one of the few places I stumbled across an element so I figured I'd check,

talbs replied:
Yeah, this is needed here since links at large have a border-bottom applied to them and our buttons didn't want this visually added. I'm making a PL issue with this noted (as it originated due to the conflicting rules there).

marcotuts commented on lms/static/certificates/sass/_components.scss:
```diff
@@ -0,0 +1,641 @@
+        margin-right: spacing-horizontal(small);
+
+        .src {
+            display: block;
```
NIT: display: block  is redundant here. (already in extend)

marcotuts commented on lms/static/certificates/sass/_components.scss:
```diff
@@ -0,0 +1,641 @@
+    margin-bottom: spacing-vertical(small);
+    border-bottom: rem(2) solid $cert-decorative-color-de-emphasized;
+    padding-bottom: spacing-vertical(xx-small);
+}
```
Q: I wasn't sure how this was being used exactly - Would this be assigned to non-heading html elements to act as a visual subheading?

talbs replied:
Yeah, this happened once I wrote these styles more than once in this file. I decided to abstract them out for this view. I don't think its something we'd add to the PL, but since I was adding redundant code, it made sense to make a view-specific extend. 

Cool or should I just dupe the code and avoid any extend confusion?

marcotuts replied:
That makes sense. Thanks! (No need to change anything here)

talbs commented:
@Kelketek, catching up on your questions...

> I see no Javascript files being loaded. When you told me that the social bar is static, should I take it to mean that no code has been added to activate the buttons when they display correctly?

Yes, no JavaScript has been added. My tasks have purely been HTML/CSS and focused on layout. I'm not sure how that functionality will be added in, but its not something I anticipated doing.

- - -

> I'm getting ValueError: The file 'images/bg-distinguished.png' could not be found with <pipeline.storage.PipelineCachedStorage object at 0x5686690>. when I try to collectstatic after these changes. Am I missing something?

> @talbs I looked into this further. These two files are in the repo:

> lms/static/certificates/css/main-ltr.css
> lms/static/certificates/css/main-rtl.css
> They appear to be generated files (no newlines/indentation), so I'm guessing they're not supposed to be there. They cause the paver update_assets command to fail.

The CSS you noted are ready for production and are indeed checked in on purpose. They are standalone to certificates and are compiled outside of our pipeline.

Sorry for the collectstatic trouble, I believe I need to add in the CSS files to our static pipeline (per @feanil's comments in person). I'll do that. I'm not sure how to handle the background image, but I'll try to sync that up too.

I'm not sure how to handle issues you're seeing with the ``paver update_assets`` command. Let's see if its still a problem once I add these CSS files to the pipeline/collectstatic list.

marcotuts commented on lms/static/certificates/sass/_components.scss:
```diff
@@ -0,0 +1,641 @@
+    }
+}
+
+// certificate - distinguished
```
NIT: comment should be "-base" here.

marcotuts commented on lms/static/certificates/sass/_layouts.scss:
```diff
@@ -0,0 +1,324 @@
+
+// message to user-centric banner
+.wrapper-banner-user {
+    margin-top: -(spacing-vertical(base)); // abutt the global header, plz
```
NIT: Too much "T" on that abutment. You can get away with one less T, Mr. http://bit.ly/1PdZ8E0

mattdrayer commented:
Jenkins, test this please.

marcotuts commented:
Other notes: 
1.) We may want to exclude font file formats we don't need. The performance team has found that our font file imports are a substantial fraction of the page load time for pages on edx-platform. They've suggested dropping eot (for IE8 and lower) and sv (iOS4 and lower).

2.) FED looked great! Lots of new pattern library stuff to absorb. :+1: 

3.) If it makes sense to review this again in a sandbox prior to merge I'm happy to do that, but for now my review is complete. 

4.) I did not review in depth the pattern library base files, since you and Chris have likely reviewed this separately. Let me know if you'd like a look through those files as well, I'm happy to do that.

marcotuts commented:
I forgot one of my comments in the list above, adding now: 

5.) We might want to add fake signature files that belong to no real human for our openedX templates. I mention this just in case the signatures currently used belong to someone.

talbs commented:
@marcotuts, thanks for the look through. I've addressed your feedback, here are details:

> We may want to exclude font file formats we don't need 

Done! I've added a GH issue for the PL to do this as well

> If it makes sense to review this again in a sandbox prior to merge I'm happy to do that, but for now my review is complete.

Once we have some of the rendering abstracted out and config set up, definitely!

> We might want to add fake signature files that belong to no real human for our openedX templates.

Done as well! This has fake sigs you guys help create

- - -

Also, I've synced up the share actions in the banner UI to mirror what we're expecting for V1. Here's a screengrab:

![screen shot 2015-05-11 at 5 41 53 pm](https://cloud.githubusercontent.com/assets/163763/7575712/110574d2-f805-11e4-9832-b3b5dc8da78d.png)

- - -

Let me know if you have any other concerns from a UX/FED review PoV.

talbs commented on static/edx-pl/sass/main-ltr.scss:
```diff
@@ -0,0 +1,22 @@
+// ------------------------------
```
@ormsbee and @feanil, **quick gut check needed**: I've moved a static version of the edX Pattern Library (https://github.com/edx/ux-pattern-library) here for web certs to use now (and down the road future things to leverage). I know we need to cover packaging for easier maintenance and versioning, but for now:

* is this alright by you?
* is this the right location (a global static directory) for this set of UI architecture?

feanil replied:
This is the SCSS of the pattern library, does this mean we expect to use the pipeline to compile it to CSS and then use that?

talbs replied:
@feanil, good question, for now these main-ltr/rtl.scss files aren't used or referenced directly. Instead the Web Certs UI has its own app-centric compile (see https://github.com/edx/edx-platform/blob/talbs/certs-web-ui/lms%2Fstatic%2Fcertificates%2Fsass%2Fmain-ltr.scss and https://github.com/edx/edx-platform/blob/talbs/certs-web-ui/lms%2Fstatic%2Fcertificates%2Fsass%2Fmain-rtl.scss) and imports a lot of these same partials noted here. The output CSS (which is compiled to https://github.com/edx/edx-platform/blob/talbs/certs-web-ui/lms%2Fstatic%2Fcertificates%2Fcss%2Fmain-ltr.css and https://github.com/edx/edx-platform/blob/talbs/certs-web-ui/lms%2Fstatic%2Fcertificates%2Fcss%2Fmain-rtl.css) seem like the files we need to track from a pipeline/collectstatic PoV, right?

(Apologies for any confusion, I just pointed you two to this file since it had a lot of references to the partials I've added and the app is using.)

talbs replied:
Also, @feanil, the collectstatic question above is topic I pinged you about today in the devops HipChat room, FYI. Just want to make sure I'm defining the new style bundle in the right place.

talbs commented:
Hi, all. Here's a bit of status on this work from my perspective:

* with the help of @feanil, the production-ready assets (CSS and images) are now being accounted for with collectstatic.
* with the help of @sarina, I've got the styling and mechanics set for RTL support (including very simple RTL support using our new grid framework [susy](http://susy.oddbird.net/). @mattdrayer, there may still be some configuration/proofing needed later once we're not relying on a URL parameter to produce a student's certificate view.
* the share actions have been synced up with what's expected for V1 (Mozilla Open Badges Backpack + Print)
* font face files have been trimmed down to just the file formats we need to support modern browsers.

- - -

Unless otherwise needed, I'll be hands off on this work until @mattdrayer has gotten further along with template abstraction. If you folks have any questions or issues, please let me know here. Thanks for the support!

talbs commented:
Also, I've squashed all of my iteration-based commits into one, so please pull and rebase if you're actively working against this.

sarina commented:
Checking your head into the repo? That's a bold move, Cotton! Let's hope it
pays off.

On Tue, May 12, 2015 at 10:02 PM, Brian Talbot <notifications@github.com>
wrote:

> Also, I've squashed all of my iteration-based commits into one, so please
> pull and rebase if you're actively working against this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/7877#issuecomment-101482084>.
>

talbs commented:
@sarina, mind giving this a local test walkthrough quickly?

talbs commented:
Some additional status: 

* I've hidden the mozilla openbadges share link within the banner we're showing an owner of the achievement for the time being (per @Kelketek's request)
* I've added some very very simple JS to handle the print button we're showing an owner of the achievement

sarina commented:
@talbs my development environment is borked so I'm having trouble testing this. however it looks like there may indeed be a compilation error based on the way your [bok choy shards are failing](https://build.testeng.edx.org/job/edx-platform-test-subset/61801/console). Can you try running `paver test_bokchoy` locally?

Kelketek commented:
@sarina @talbs Bok_choy tests will not pass until https://github.com/edx/edx-platform/pull/8058 is merged or an alternate solution is found, if that's not acceptable for some reason.

talbs commented:
@sarina, mind taking a look at #8058 to see if that solves things in a way we're cool with (its beyond my expertise)?

sarina commented:
@talbs I thin it's a bit beyond my expertise as well. All I can tell is the bok choy builds on this branch aren't starting up. They clearly run & pass on #8058 though.  There's a lot going on in both PRs that I can't quite follow.

Do we have an owner for this stuff (pipeline, specifically) that we could perhaps ping for review?

talbs commented:
Hmm, I think @feanil might be our best bet for the pipeline stuff on #8058 (or someone he recommends). Sorry to keep passing the buck.

sarina commented:
OK - will just lodge it here that we shouldn't check in any university-branded assets into the platform.

My concern about a directory named "edx-pl" is it's unclear what it's for. We use "edx-theme" as a directory name in the lms & cms to denote edx branded images (eg `lms/static/images/edx-theme/`). My concern with "edx-pl" it's unclear if they're images specific to edX (as in edx.org, and the edx brand), or if they're generally applicable to the entire platform. I'd advocate for using either "edx"/"edx-theme" if they're all edx-specific images, or something like "platform" or "edx-platform" if it's intended to be images applicable for any instance of edx-platform.

talbs commented:
Thanks, @sarina. I've 1) removed the partner logos used for UI testing (and replaced the references with our standard placeholder platform logo)  and 2) renamed the ``edx-pl`` directory to ``pattern-library`` for clarity.

sarina commented:
:+1:

On Tue, May 19, 2015 at 3:41 PM, Brian Talbot <notifications@github.com>
wrote:

> Thanks, @sarina <https://github.com/sarina>. I've 1) removed the partner
> logos used for UI testing (and replaced the references with our standard
> placeholder platform logo) and 2) renamed the edx-pl directory to
> pattern-library for clarity.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/7877#issuecomment-103647717>.
>

talbs commented:
Given the shift in how to get this work merged into on-going development work, I'm closing this PR in favor of https://github.com/edx/edx-platform/pull/8106, which merges things into the SOL-465 branch of work.

Shout if you folks have any concerns.
